File: src/py/pe_0003.py
# FUNCTIONS


def prime_factors(n):
    i = 2
    out = tuple()
    while n != 1:
        if n % i == 0:
            n //= i
            out += i,
        else:
            i += 1
    return out


# SCRIPT


if __name__ == "__main__":
    assert prime_factors(13195) == (5, 7, 13, 29)
    print(max(prime_factors(600851475143)))

File: src/c/pe_0003.c
#include <stdio.h>


// SCRIPT


int main() {
    long long int n, i;

    n = 600851475143L;
    i = 2;

    while (n != 1) {
        if (n % i == 0) {
            n = n / i;
        } else {
            i = i + 1;
        }
    }

    printf("%d\n", i);
}
